Documentation/tutorial/howto_add_a_customized_post_processor.md

```diff
@@ -0,0 +1,78 @@
+How-to: Add a customized post-processor
+====================================
+
+We would have the ability to process output HTML files by adding a customized post-processor.
+In DocFX, the index file for full-text-search is generated by one post-processor named `ExtractSearchIndex`.
+In this topic, we will show how to add a customized post-processor.
+
+## Step0: Preparation
+
+* Create a new C# class library project in `Visual Studio`.
+* Add nuget packages:
+    * `System.Collections.Immutable` with version 1.1.37
+    * `Microsoft.Composition` with version 1.0.27
+* Add `Microsoft.DocAsCode.Plugins`
+If build DocFX from source code, add reference to the project.
+Otherwise, add nuget package `Microsoft.DocAsCode.Plugins` with the same version of DocFX.
+
+## Step1: Create a new class (MyProcessor.cs) with following code:
+
+```csharp
+[Export(nameof(MyProcessor), typeof(IPostProcessor))]
+public class MyProcessor : IPostProcessor
+{
+    // TODO: implements IPostProcessor
+}
+```
+
+## Step2: Update global metadata
+
+```csharp
+public ImmutableDictionary<string, object> UpdateMetadata(ImmutableDictionary<string, object> metadata)
+{
+    // TODO: add/remove/update property from global metadata
+    return metadata;
+}
+```
+
+In this method, we can update the global metadata before building all the files declared in `docfx.json`. Otherwise, you can just return the metadata from parameter if you don't need to change global metadata.
+
+Take `ExtractSearchIndex` for example, we add `"_enableSearch": true` in global metadata. Then, the default template would know it should load search box in navbar.
+
+## Step3: Process all the HTML files generated by DocFX
+
+```csharp
+    public Manifest Process(Manifest manifest, string outputFolder)
+    {
+        // TODO: add/remove/update all the HTML files included in manifest
+        return manifest;
+    }
+```
+
+In this method, we can get all HTML files path using `manifest` and save some files using `outputFolder` which is the destination our static website would exist. Then, operations can be added to process all the HTML files generated by DocFX.
+
+Take `ExtractSearchIndex` for example again, we traverse all HTML files, extract key words from these HTML files and save a file named `index.json` under the `outputFolder`. At last, we return the manifest which is not modified.
+
+## Step4: Build your project and copy the output dll files to:
+
+* Global: the folder with name `Plugins` under DocFX.exe
+* Non-global: the folder with name `Plugins` under a template folder, then run `DocFX build` command with parameter `-t {template}`.
+
+    *Hint*: DocFX can merge templates, that means create a template only contains `Plugins` folder, then run command `DocFX build` with parameter `-t {templateForRender},{templateForPlugins}`.
```

Comment:
that means create a template only contains `Plugins` folder, then run command `DocFX build` with parameter `-t {templateForRender},{templateForPlugins}` ->
so we can specify multiple template folders as `DocFX build -t {templateForRender},{templateForPlugins}`. Each of template folder should have a subfolder named `Plugins` with exported assemblies.